### PR TITLE
Update CI helpers

### PR DESCRIPTION
**Pull request by rikhuijzer**
Add CompatHelper and update TagBot. 

I'm not sure whether CompatHelper is actually a good addition to Example.jl. Let me know if it should be removed.

I see that CI for tests is worked on in https://github.com/JuliaLang/Example.jl/pull/55, so I am not touching the CI for tests.

### CODE REVIEW

**Comment by rikhuijzer** [comment]
Apologies for the weird commit message, by the way. I'm fiddling with some automation and that line was in the example configuration. Feel free to squash the PR to get rid of the weird commit message.

**Review by nalimilan — APPROVED** [review]

File: .github/workflows/CompatHelper.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+name: CompatHelper
+on:
+  schedule:
+    - cron: 0 0 * * *
+  workflow_dispatch:
+jobs:
+  CompatHelper:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Pkg.add("CompatHelper")
```
Comment (KristofferC) [inline .github/workflows/CompatHelper.yml:10]: Isn't it better to collapse these two steps into one so Julia only has to be started one time?

Comment (rikhuijzer) [reply .github/workflows/CompatHelper.yml:10]: It is like this at https://github.com/JuliaRegistries/CompatHelper.jl, so I guess Dilum Aluthge has good reasons for it. My guess is for readability reasons.

Comment (rikhuijzer) [reply .github/workflows/CompatHelper.yml:10]: By the way, I generated this file from PkgTemplates. As a fun project, I'm planning to run a weekly CI job to compare the generated PkgTemplates files with Example.jl.

**Comment by DilumAluthge** [comment]
I would remove CompatHelper, which at some point in the future we will deprecate in favor of Dependabot.

**Comment by rikhuijzer** [comment]
I cannot make changes to this PR anymore because I removed the fork. Closing this PR.

**Comment by rikhuijzer** [comment]
> I would remove CompatHelper, which at some point in the future we will deprecate in favor of Dependabot.

Good to know still. Thanks

**Pull request closed without merging**